browser/ui/webui/brave_rewards_page_ui.cc

```diff
@@ -485,6 +506,12 @@ void RewardsDOMHandler::Init() {
   rewards_service_->StartProcess(base::DoNothing());
 
   ads_service_ = brave_ads::AdsServiceFactory::GetForProfile(profile);
+
+  profile_pref_change_registrar_.Init(profile->GetPrefs());
+  profile_pref_change_registrar_.Add(
```

Comment:
this should be unregistered in `OnJavascriptDisallowed`

Reply:
also it shouldn't be registered here, it should be registered in `OnJavascriptAllowed`

Reply:
resolved - https://github.com/brave/brave-core/pull/8566/files#diff-41756d127f52e09eaa4d10553f84a5a4f6b8fd674c623ec67015a367e4f79f93R82